mopidy/__main__.py

```diff
@@ -12,13 +15,21 @@
 
 try:
     # Make GLib's mainloop the event loop for python-dbus
-    import dbus.mainloop.glib
+    import dbus.mainloop.glib  # pyright: ignore[reportMissingImports]
 
     dbus.mainloop.glib.threads_init()
     dbus.mainloop.glib.DBusGMainLoop(set_as_default=True)
 except ImportError:
     pass
 
+if TYPE_CHECKING:
+
```

Comment:
Please allow a comment.

Do you really want all these “if TYPE_CHECKING”

in my experience adding the class just for type checking is bound to create problem later, it would be better just to have the definition/implementation once and import it.

If that causes a circular import, then normally it’s a design problem.

Reply:
I'm totally in on removing the `if` in this case at least. I see the typed dict as a stepping stone in the right direction. Longer term, I'd like most dict structures to be replaced with something like Pydantic models. In many places that cannot easily be done without changing the interface of extensions, so it will have to be done slower and with more care.

In have to admit I opened this PR the moment I got everything passing at the same time. I'll give the PR another pass and see what I'll change.

Reply:
You could also just merge this (with the review you requested) and then do the changes later, which would allow e.g. me to help (if wanted). I am here for the long run, with time to spare so feel free to push work that you find boring in my direction.

I am working on removing pkg_resources and I try to add typing when I change things,... I need to have multiroom functionality using my mini mac etc, and to do that The kernel/internal needs some extensions, so I know I have to learn a lot and hope you guys do not loose patience.